primitives/src/authorization.rs

```diff
@@ -0,0 +1,41 @@
+use crate::identity_id::IdentityId;
+use codec::{Decode, Encode};
+use rstd::prelude::Vec;
+
+/// Authorization data for two step prcoesses.
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Debug, PartialOrd, Ord)]
+pub enum AuthorizationData {
+    /// Authorization to transfer a ticker
+    TransferTicker(Vec<u8>),
+    /// Any other authorization
+    Custom(Vec<u8>),
+    /// No authorization data
+    None,
+}
+
+impl Default for AuthorizationData {
```

Comment:
This instance would have been trivial if `Option` were used.

Reply:
True but it doesn't really gain us anything other than saving a few lines of code. `Option` is cool but it increases the size of the variable. It also forces us to unwrap data whenever we want to use it. Thus, adding to both storage and execution costs.

Unless the "default" value represents something other than "none", I avoid using Option. For example, you'll see below that I'm using 1-indexed auth_id/auth_number rather than 0-index below because I am representing "none" as 0. I could have used Option<u64> and then used 0-index auth_id but I would have gained nothing from it except maybe a little more code clarity.

At the same time, I am using Option<expiry> because 0 can mean something other than none when it comes to timestamps. I don't think anyone will use 0 timestamps in real life but they can to show something historic that isn't valid anymore.

Basically, If I can control the data (enum/ids), I prefer to not use option and just use one specific data to represent none. However, if the data is user input, I prefer to use Option because in that case, I can not be sure that the user means 0 or none when they enter 0.

Reply:
I see your point. I would personally prefer code clarity because this all fits into a common use case for option types.

> Unless the "default" value represents something other than "none", I avoid using Option.

I misread this at first and thought you were referring to manual `Default` instances. If you instead mean `0` vs `None` then I agree: `Timestamp(0)` is not the same as `None: Option<Timestamp>`.